### ISSUE TITLE
OpenAPI string uri format

### ISSUE THREAD

**Issue by rubensa** [stoplightio/spectral#2484]
**Describe the bug**

The **uri** format for the **string** type in OpenAPI is not validated correctly.  Looks like only absolute URIs (vs relative) are supported.

The [URI specification says](https://www.rfc-editor.org/rfc/rfc3986#appendix-B):

```
The following line is the regular expression for breaking-down a well-formed URI reference into its components.

^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?
```

**To Reproduce**

1. Given this OpenAPI document

```yaml
openapi: 3.0.0
info:
  contact:
    name: Test
  description: Test
  title: Test
  version: 1.0.0
servers:
  - url: http://localhost:8080
    description: Localhost
paths:
  /test:
    get:
      description: Test
      operationId: test
      responses:
        "200":
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  instance:
                    type: string
                    format: uri
                example: { "instance": "/tenant/api/v1/scenario" }
      tags:
        - test
tags:
  - name: test
    description: Test
```

2. See error

`"instance" property must match format "uri"`

**Expected behavior**

No errors are shown as the provided _instance_ URI value is correct according to the [URI specification](https://www.rfc-editor.org/rfc/rfc3986).

**Screenshots**

![Screenshot from 2023-06-09 11-26-44](https://github.com/stoplightio/spectral/assets/1469340/8192ac54-12e7-4bb8-84cf-65b038bb7888)

**Environment (remove any that are not applicable):**
 - Library version: The one included with the [Spectral Linter for VS Code](https://github.com/stoplightio/vscode-spectral) v1.1.2
 - OS: Ubuntu 22.04

**Additional context**

If I replace:
```yaml
                  instance:
                    type: string
                    format: uri
```
with:
```yaml
                  instance:
                    type: string
                    pattern: '^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?'
```
format validation success.

**Comment by LasneF** [comment]
my guess is that it falls inside the pitfall describe here 
https://json-schema.org/understanding-json-schema/reference/string.html#resource-identifiers 
"
If the values in the schema have the ability to be relative to a particular source path (such as a link from a webpage), it is generally better practice to use "uri-reference" (or "iri-reference") rather than "uri" (or "iri"). "uri" should only be used when the path must be absolute.

Draft-specific info:
[Draft 4](https://json-schema.org/understanding-json-schema/reference/string.html#7f2353e1f4d0_Draft_4)
Draft 4 only includes "uri", not "uri-reference". Therefore, there is some ambiguity around whether "uri" should accept relative paths.
" 

so here there is not 100% clarity , only advise would be to set uri-reference (tested and validated with your code )

**Comment by rubensa** [comment]
Thanks for the info @LasneF, didn't know about that.

I close the issue as it works using `uri-reference` instead of `uri` as **format**.
